docs/interop.md

```diff
@@ -0,0 +1,69 @@
+The right to spend a token is enforced according to the conditions within the script.
+
+## HTLC (atomic cross-chain swap or exchange)
+
+HTLC (Hash Time Locked Contract) is a token transfer that use hashlocks and timelocks to require that the recipient of a token either acknowledge receiving the token prior to a deadline by generating cryptographic proof or forfeit the ability to claim the token, returning it to the sender.
+With this mechanism Token SDK supports atomic cross-chain swap or exchange of tokens. 
+
+### HTLC script
+
+The HTLC (exchange) script encodes the details of the exchange, the identities of the sender and the recipient of the token, a deadline, and hashing information.
+The hashing information includes the hash itself, the hash function used (e.g., SHA-256), and the encoding (e.g., Base64).
+The hash is chosen by the sender and the recipient must provide the preimage for the exchange to happen.
+
+```go
+// Script contains the details of an exchange
+type Script struct {
+    Sender    view.Identity
+    Recipient view.Identity
+    Deadline  time.Time
+    HashInfo  HashInfo
+}
+
+// HashInfo contains the information regarding the hashing
+type HashInfo struct {
+    Hash         []byte
+    HashFunc     crypto.Hash
+    HashEncoding encoding.Encoding
+}
+```
+
+## Interoperability services
+
+The interoperability services which are responsible for assembling the token transaction, managing the transaction lifecycle, and so on, are the same as the [`Token Transaction Services`](./services.md). 
+They are located in `token/services/interop`.
+
+Other services are script specific.
+For example, the token transaction assembling service enables appending exchange, claim, or reclaim actions to the token request of the transaction. All of these actions translate into a transfer action. 
+Exchange is the "locking" process, where the sender sets the details of the exchange and transfers ownership of the token to a script. 
+Claim allows the recipient to gain ownership of the token by providing the preimage. 
+Reclaim returns the token to the sender. 
+Claim must happen before the deadline ends, while reclaim can only occur after the deadline has passed.
+
+```go
+func (t *Transaction) Exchange(wallet *token.OwnerWallet, sender view.Identity, typ string, value uint64, recipient view.Identity, deadline time.Duration, opts ...token.TransferOption) ([]byte, error)
+func (t *Transaction) Claim(wallet *token.OwnerWallet, tok *token2.UnspentToken, preImage []byte) error
+func (t *Transaction) Reclaim(wallet *token.OwnerWallet, tok *token2.UnspentToken) error
+```
+
+The interop `Wallet` service, located in `token/services/interop/wallet.go`, supports listing expired tokens, whose deadline have passed, and listing tokens with a desired matching preimage.
+
+In addition, the interop signer and verifier services are script specific, for example in the HTLC case the preimage is part of the signed message.
+
+## Driver adjustments 
```

Comment:
Shall we move this to the specific driver implementations (`docs/fabtoken.md`, `docs/zkat-dlog.md`) as updated by PR #351 ?

Reply:
In my opinion it should be together with the interop explanations and most of it is the same for both drivers.

Reply:
I suggest to just link to the docs for more details on the zkat-dlog driver impl.